### PR TITLE
Namedtuple return for symeig, eig, pstrf, qr, geqrf

### PR DESCRIPTION
**Pull request by zasdfgbnm**
More ops for https://github.com/pytorch/pytorch/issues/394

cc: @ezyang

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
test not added yet

**Comment by zasdfgbnm** [comment]
Should be working now

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -3583,21 +3583,21 @@
   matches_jit_signature: True
   variants: method, function
 
-- func: pstrf(Tensor self, bool upper=True, Scalar tol=-1, *, Tensor(a!) u, Tensor(b!) piv) ->(Tensor(a!), Tensor(b!))
+- func: pstrf(Tensor self, bool upper=True, Scalar tol=-1, *, Tensor(a!) u, Tensor(b!) piv) ->(Tensor(a!) u, Tensor(b!) piv)
```
Comment (zou3519) [inline aten/src/ATen/native/native_functions.yaml]: @vishwakftw I remember you were interested in cholesky decompositions, what do you think about these names?

Personally I have no clue what these names are but I also don't know anything on the subject

**Comment by ezyang** [comment]
@zasdfgbnm what's your opinion on name conflicts between input and output arguments? An alternate resolution is to say that, the output name always has to be distinct from input names (and then we have to pick a name like `e` and `V`, which I guess is worse than `eigenvalues` and `eigenvectors` for non-mathematicians, but is not totally unprecedented, since we have a bunch of other very shortly named identifiers.)

**Comment by ezyang** [comment]
This PR looks functionally correct but I want to discuss the matter of `_return` some more

**Comment by zasdfgbnm** [comment]
@ezyang I am OK with short names in general, like `Q`, `R`, but personally I don't like the name `e` and `V` for this one. Because the `e` and `V` are not as widely adopted as something like `Q` and `R` for QR decomposition. IMHO, it's good to use abbreviations only when there is a common way to abbreviate. 

I don't thin there is a common way to abbreviate eigenvalues and eigenvectors, below are a list of abbreviations in different materials (these are all top ranking result in Google when searching `eigen values`):
- lambda and X: http://mathworld.wolfram.com/Eigenvalue.html
- lambda and v: https://en.wikipedia.org/wiki/Eigenvalues_and_eigenvectors
- lambda and eta: http://tutorial.math.lamar.edu/Classes/DE/LA_Eigen.aspx

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (vishwakftw) [reply aten/src/ATen/native/native_functions.yaml]: In this case, `u` seems right, because `pstrf` is meant to return upper triangular matrices.

**Comment by ezyang** [comment]
SGTM. @soumith suggests we use `pivot` instead of `piv` though :)

**Comment by zasdfgbnm** [comment]
@ezyang Just renamed piv -> pivot

**Comment by zasdfgbnm** [comment]
Need to rebase after landing https://github.com/pytorch/pytorch/pull/16186, because we need to update the whitelist of the new unit test added in https://github.com/pytorch/pytorch/pull/16186.

**Review by ezyang — APPROVED** [review]
ok

**Comment by zasdfgbnm** [comment]
landed

**Comment by zasdfgbnm** [comment]
@ezyang Please review https://github.com/pytorch/pytorch/pull/17093 before https://github.com/pytorch/pytorch/pull/17195. As the more and more operators returning structseq, the test `test_namedtuple_return` becomes long. I will do a refactor on `test_namedtuple_return ` in #17195  to reduce its size.

**Pull request closed without merging**